### PR TITLE
Fix error and depwarn on 0.7

### LINKED ISSUE #42: StackOverflow in v0.7
Just in case you didn't spot this on Travis nightly:

```
ERROR: LoadError: LoadError: StackOverflowError:
Stacktrace:
 [1] include_from_node1(::Module, ::String) at ./loading.jl:549
 [2] include(::Module, ::String) at ./sysimg.jl:14
 [3] include(::String) at /home/travis/.julia/v0.7/MacroTools/src/MacroTools.jl:2
 [4] include_from_node1(::Module, ::String) at ./loading.jl:549
 [5] include(::Module, ::String) at ./sysimg.jl:14
 [6] anonymous at ./<missing>:2
while loading /home/travis/.julia/v0.7/MacroTools/src/utils.jl, in expression starting on line 96
while loading /home/travis/.julia/v0.7/MacroTools/src/MacroTools.jl, in expression starting on line 12
```

I didn't know I was using this, but I've added a Juno.jl dependency so it looks like I am... :)

### PR DESCRIPTION
**Pull request by yuyichao**
Fix #42

### CODE REVIEW

**Comment by MikeInnes** [comment]
Thanks!

File: src/utils.jl [hunk 1]
```diff
@@ -155,7 +145,7 @@ More convenient macro expansion, e.g.
     @expand @time foo()
 """
 macro expand(ex)
-  :(alias_gensyms(macroexpand($(ex,)[1])))
+  :(alias_gensyms(macroexpand($(@__MODULE__), $(ex,)[1])))
```
Comment (yuyichao) [inline src/utils.jl]: actually I just relize that this might be wrong...

Comment (MikeInnes) [inline src/utils.jl]: I think this is what @vtjnash suggested in the original issue – I'm not up to date on all the macro changes myself though.

Comment (vtjnash) [inline src/utils.jl]: Either splice in `$__module__`, or expand `@__MODULE__`, don't splice in `@__MODULE__`

**Pull request merged**